### PR TITLE
load: load should not need metadata

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #1320 

### What is changed and how it works?

since full mode doesn't not need REPLICATION related privilege and doesn't check it, mydumper/dumpling won't generate a valid `metdata` file containing "SHOW MASTER STATUS:" at then. And when loader parse it, load task will stop.

Now load should ignore that error

### Check List 

Tests 

 - pass original test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
Why didn't relay unit start sync from metadata's position, thus we need not `unitTransWaitCondition` 🤔

**Comment by GMHDBJD** [comment]
> Why didn't relay unit start sync from metadata's position, thus we need not `unitTransWaitCondition` 🤔

I think relay should be source level but not task level. So it may start from now, or start from pos/gtid which we specify

File: loader/loader.go [hunk 1]
```diff
@@ -1307,6 +1307,11 @@ func (l *Loader) getMydumpMetadata() error {
 	metafile := filepath.Join(l.cfg.LoaderConfig.Dir, "metadata")
 	loc, _, err := dumpling.ParseMetaData(metafile, l.cfg.Flavor)
 	if err != nil {
+		if terror.ErrMetadataNoBinlogLoc.Equal(err) {
+			l.logger.Warn("dumped metadata doesn't have binlog location, it's OK if DM doesn't enter incremental mode")
```
Comment (GMHDBJD) [inline loader/loader.go:1311]: How about check task_mode here? 
It is annoying to report errors after a long time import is completed

Comment (lance6716) [reply loader/loader.go:1311]: We're checking REPLICATION CLIENT in pre-check wrongly, (should make sure it's grants on `*.*`). So for all mode and incremental mode, pre-check could make sure there's enough privilege generating correct metadata. For full mode, it's no harm

Comment (GMHDBJD) [reply loader/loader.go:1311]: Are there any other cases that may cause this error? 🤔

**Review by GMHDBJD — COMMENTED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**